src/main/java/repository/RepositoryImpl.java

```diff
@@ -0,0 +1,52 @@
+package repository;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class RepositoryImpl implements Repository {
+    private static Repository repositoryImpl;
+
+    private static final Map<String, Map<String, String>> REPOSITORY_DATA = new ConcurrentHashMap<>();
+
+    private RepositoryImpl() {
+    }
+
+    public static Repository getInstance() {
+        if (repositoryImpl == null) {
+            return new RepositoryImpl();
+        }
+        return repositoryImpl;
+    }
+
+    public void putNewData(String key) {
+        REPOSITORY_DATA.put(key, new ConcurrentHashMap<>());
+    }
+
+    public void update(String sessionID, String parameterName, String paramValue) {
```

Comment:
Why does this paramter **sessionID**? Repository should not know about any session. It is "id" like database primary key.